src/main/java/org/jabref/gui/importer/ImportEntriesViewModel.java

```diff
@@ -106,12 +119,26 @@ public boolean hasDuplicate(BibEntry entry) {
     public void importEntries(List<BibEntry> entriesToImport, boolean downloadFiles) {
         // Check if we are supposed to warn about duplicates.
         // If so, then see if there are duplicates, and warn if yes.
+        this.entriesToImport = entriesToImport;
         if (preferences.shouldWarnAboutDuplicatesForImport()) {
             BackgroundTask.wrap(() -> entriesToImport.stream()
                                                      .anyMatch(this::hasDuplicate)).onSuccess(duplicateFound -> {
                 if (duplicateFound) {
+                    StringBuffer duplicateInfo = new StringBuffer("There are possible duplicates (") ;
+                    if (this.datasetDuplicate.isPresent()){
+                        duplicateInfo.append("duplicate to the entry in the database whose Bibtexkey is : ");
+                        duplicateInfo.append(this.datasetDuplicate.get().getField(InternalField.KEY_FIELD).get());
+                    }
+                    else if (this.internalDuplicate.isPresent()){
+                        duplicateInfo.append("duplicate to the entry in the list of entries to be imported whose title is : ");
+                        duplicateInfo.append(this.internalDuplicate.get().getField(StandardField.TITLE).get());
+
+                    }
+                    duplicateInfo.append(") that haven't been resolved. Continue?");
+
                     boolean continueImport = dialogService.showConfirmationDialogWithOptOutAndWait(Localization.lang("Duplicates found"),
-                            Localization.lang("There are possible duplicates (marked with an icon) that haven't been resolved. Continue?"),
+                            //Localization.lang("There are possible duplicates (marked with an icon) that haven't been resolved. Continue?"),
+                            Localization.lang(duplicateInfo.toString()),
```

Comment:
This is unfortunately not how localization works. The string has to be directly passed to `Localization.lang()` as plain string. Use `%0` as place holder.

Reply:
In my opinion, there are two cases that we need consider. First, the duplicate entry is in database. I think showing it's bibtexkey can help use find it quickly. Second, the duplicate entry is in the list entries that want to be imported. I think showing title is a better way because their bibtexkey is not shown in the import dialog. In a result, I have to concatenate string to meet different cases. So, do you have suggestion for this problem? Do I need revert this change?